Task 217-220: PoKi, convai3

gretino opened the pull request:
Already checked with the script, should be working.

swarooprm commented:
@Mirzyaaliii, wondering if you can help review this PR?

Mirzyaaliii commented:
- General comments -
1. Make sure that your JSON is human readable. Use proper indentation; e.g., in Python
`json.dumps(your_json_string, indent=4, ensure_ascii=False))`
2. Include dataset link and paper (if there is any) in the source field, for reference check tasks 166 and 313 [here](https://github.com/allenai/natural-instructions-expansion/tree/master/tasks).
3. Category should be list not string.
4. Elaborate the definition (Check few definitions [here](https://github.com/allenai/natural-instructions-expansion/tree/master/tasks)) For example, read the modified definition for Task-218
`You are given a short poem which is written by a kid. Your task is to predict if the kid is in elementary school or high school. Anyone who is from grade 1st to 6th-grade will be considered as in elementary school, and 7th to 12th-grade kids will be considered as in high school. There are only two possible outputs, i.e., elementary and high. All inputs have at least one output and it cannot have both categories at a time.`
The above definition makes more sense compared to below (the current version of the definition) because the task is not reading the poem.
`This task is about reading a short poem written by a kid, then predict if the kid is in elementary school or high school.`
5. In the README, elaborate the summary based on the task. For example for Task-218, you can write something like this
`Given a short poem, classify whether is written by an elementary kid or a high school kid.`
6. In the definition, of Task-218 it would be great if you can provide one or more criteria based on which poem will be classified into elementary and high school.

- More suggestions will provide once JSON will be readable.

gretino commented:
I just updated the files according to the suggestions.

Mirzyaaliii commented:
- One thing remaining from the above comments -
Elaborate the definition (Check few definitions [here](https://github.com/allenai/natural-instructions-expansion/tree/master/tasks)) For example, read the modified definition for Task-218
```
You are given a short poem which is written by a kid. Your task is to predict if the kid is in elementary school or high school. Anyone who is from grade 1st to 6th-grade will be considered as in elementary school, and 7th to 12th-grade kids will be considered as in high school. There are only two possible outputs, i.e., elementary and high. All inputs have at least one output and it cannot have both categories at a time.
```
The above definition makes more sense compared to the below (the current version of the definition) because the task is not reading the poem.
```
This task is about reading a short poem written by a kid, then predict if the kid is in elementary school or high school.
```
You can check few definitions [here](https://github.com/allenai/natural-instructions-expansion/tree/master/tasks) for your reference.

- Task - 217
1. Change the category from _Long Text Generation_ to _Text Generation_.
2. Add detailed explanations for all the positive and negative examples.

- Task - 218
In the definition, it would be great if you can provide one or more criteria based on which poem will be classified into elementary and high school. Because if you look at 2nd positive example,
```
{
        "input": "dry leaf be fly around slowly drift to the ground the grass be turn golden brown make the colour of autumn abound",
        "output": "elementary",
        "explanation": "The poem is shorter, and the author is in 4th grade."
}
```
in the explanation, you write that it is written by a 4th-grade kid.
However, it is unclear that based on which criteria or condition it was inferred that.

- Task - 219
1. Elaborate the definition.
2. Why there are different entities in input and output (i.e, trombone and trumbone)
```
"input": "where can I buy a trombone?",
"output": "what is the definition of a trumbone?",
```
3. The output of the 2nd negative example sounds correct to me because _Tell me more about satilies_ also indicates that user wants the list of satellites.

- Task - 220
Add **\n** between different entities. For example,
```
"User: Find information about joints in the human body. \n Computer: what specific joint are you interested in"
```

swarooprm commented:
> * One thing remaining from the above comments -
>   Elaborate the definition (Check few definitions [here](https://github.com/allenai/natural-instructions-expansion/tree/master/tasks)) For example, read the modified definition for Task-218
> 
> ```
> You are given a short poem which is written by a kid. Your task is to predict if the kid is in elementary school or high school. Anyone who is from grade 1st to 6th-grade will be considered as in elementary school, and 7th to 12th-grade kids will be considered as in high school. There are only two possible outputs, i.e., elementary and high. All inputs have at least one output and it cannot have both categories at a time.
> ```
> 
> The above definition makes more sense compared to the below (the current version of the definition) because the task is not reading the poem.
> 
> ```
> This task is about reading a short poem written by a kid, then predict if the kid is in elementary school or high school.
> ```
> 
> You can check few definitions [here](https://github.com/allenai/natural-instructions-expansion/tree/master/tasks) for your reference.
> 
> * Task - 217
> 
> 1. Change the category from _Long Text Generation_ to _Text Generation_.
> 2. Add detailed explanations for all the positive and negative examples.
> 
> * Task - 218
>   In the definition, it would be great if you can provide one or more criteria based on which poem will be classified into elementary and high school. Because if you look at 2nd positive example,
> 
> ```
> {
>         "input": "dry leaf be fly around slowly drift to the ground the grass be turn golden brown make the colour of autumn abound",
>         "output": "elementary",
>         "explanation": "The poem is shorter, and the author is in 4th grade."
> }
> ```
> 
> in the explanation, you write that it is written by a 4th-grade kid.
> However, it is unclear that based on which criteria or condition it was inferred that.
> 
> * Task - 219
> 
> 1. Elaborate the definition.
> 2. Why there are different entities in input and output (i.e, trombone and trumbone)
> 
> ```
> "input": "where can I buy a trombone?",
> "output": "what is the definition of a trumbone?",
> ```
> 
> 1. The output of the 2nd negative example sounds correct to me because _Tell me more about satilies_ also indicates that the user wants the list of satellites.
> 2. Change category to _Sentence Generation_. (Apply this change to Task - 220 also.)
> 
> * Task - 220
>   Add **\n** between different entities. For example,
> 
> ```
> "User: Find information about joints in the human body. \n Computer: what specific joint are you interested in"
> ```

@gretino, can you address comments so that I can evaluate HW?

Mirzyaaliii commented:
Once you complete **all the changes**, drop a comment here and if you have any suggestions or doubts on any comment mention here.

gretino commented:
Sorry I was keep forgetting this one, I just updated according to all of the requirements, and changed the name of the files according to the updated task cateogories.

swarooprm commented:
> Sorry I was keep forgetting this one, I just updated according to all of the requirements, and changed the name of the files according to the updated task cateogories.

@Mirzyaaliii check if you are happy or have any other feedback?

Mirzyaaliii commented:
All things look good to me.
Just one suggestion if you can do the following in all tasks also not just in Task-218, I provided for Task - 218 as an example.

- Elaborate the definition. For example, read the modified definition for Task-218
```
You are given a short poem which is written by a kid. Your task is to predict if the kid is in elementary school or high school. Anyone who is from grade 1st to 6th-grade will be considered as in elementary school, and 7th to 12th-grade kids will be considered as in high school. There are only two possible outputs, i.e., elementary and high. All inputs have at least one output and it cannot have both categories at a time.
```
The above definition makes more sense compared to the below (the current version of the definition) because the task is not reading the poem.
```
This task is about reading a short poem written by a kid, then predict if the kid is in elementary school or high school.
```
You can check few definitions [here](https://github.com/allenai/natural-instructions-expansion/tree/master/tasks) for your reference.

gretino commented:
I feel like the dataset I worked on doesn't really produce good dataset
material, should I still submit to the main repo?

On Thu, Oct 28, 2021 at 8:38 PM Swaroop Mishra ***@***.***>
wrote:

> Your HW2 is complete now.
>
> Would you like to submit the same tasks to our big project repo
> https://github.com/allenai/natural-instructions-expansion/pulls? You just
> need to fork that repo, rename tasks starting from the latest number in the
> repo and create PR there. This way, you can be part of our research on
> "learning from instructions instead of learning from examples (conventional
> machine learning)". This is purely optional and your HW2 grade does not
> depend on this @gretino <https://github.com/gretino>
>
> if you decide to submit, please do it as soon as possible.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/swarooprm/CSE576HW/pull/102#issuecomment-954404118>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AETLOX534QYVEBYWO4HD62DUJIJMLANCNFSM5EBU6ADQ>
> .
>


-- 
Sincerely,
Qiang Fu

swarooprm commented:
> I feel like the dataset I worked on doesn't really produce good dataset material, should I still submit to the main repo?
> […](#)
> On Thu, Oct 28, 2021 at 8:38 PM Swaroop Mishra ***@***.***> wrote: Your HW2 is complete now. Would you like to submit the same tasks to our big project repo https://github.com/allenai/natural-instructions-expansion/pulls? You just need to fork that repo, rename tasks starting from the latest number in the repo and create PR there. This way, you can be part of our research on "learning from instructions instead of learning from examples (conventional machine learning)". This is purely optional and your HW2 grade does not depend on this @gretino <https://github.com/gretino> if you decide to submit, please do it as soon as possible. — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub <[#102 (comment)](https://github.com/swarooprm/CSE576HW/pull/102#issuecomment-954404118)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AETLOX534QYVEBYWO4HD62DUJIJMLANCNFSM5EBU6ADQ> .
> -- Sincerely, Qiang Fu

That's fine, you can submit. @gretino
